server/subscribe.go

```diff
@@ -360,22 +365,95 @@ func (p *Plugin) listChannelSubscriptions() (string, error) {
 		return "", err
 	}
 
+	teamDMGMSubs, err := p.getSubscriptionsOrderedByTeamDMandGM()
+	if err != nil {
+		return "", err
+	}
+
 	rows := []string{}
+	printDMGMHeader := true
+
+	for _, teamSubs := range teamDMGMSubs {
+
+		// create team header for channels.  Only print header for DMs and GMs channels once.
+		if teamSubs.TeamId != "" {
+			team, appErr := p.API.GetTeam(teamSubs.TeamId)
+			if appErr != nil {
+				return "", appErr
+			}
+			rows = append(rows, fmt.Sprintf("\n### %s", team.DisplayName))
+		} else {
+			if printDMGMHeader {
+				rows = append(rows, fmt.Sprintf("\n### %s", "Group and Direct Messages"))
+				printDMGMHeader = false
+			}
+		}
+
+		var printChannelName = true
+		for _, subId := range teamSubs.SubIds {
+			sub := subs.Channel.ById[subId]
+
+			channel, appErr := p.API.GetChannel(sub.ChannelId)
+			if appErr != nil {
+				return "", errors.New("Failed to get channel")
+			}
+
+			// only print channel name once for all subscriptions
+			if printChannelName {
+				rows = append(rows, fmt.Sprintf("* **~%s** (%d):", channel.Name, len(teamSubs.SubIds)))
+				printChannelName = false
+			}
+
+			subName := "(No Name)"
+			if sub.Name != "" {
+				subName = sub.Name
+			}
+			rows = append(rows, fmt.Sprintf("  * %s - %s", sub.Filters.Projects.Elems()[0], subName))
+		}
+	}
+
+	return strings.Join(rows, "\n"), nil
+}
+
+func (p *Plugin) getSubscriptionsOrderedByTeamDMandGM() ([]TeamSubscriptions, error) {
```

Comment:
2/5 s/getSubscriptionsOrderedByTeamDMandGM/getSubscriptionsSorted/  - the original name is difficult to understand without the context. For now, is there another use-case where we sort the list differently?